e2e-tests/batch-mode-tests.sh

```diff
@@ -93,27 +50,8 @@ function test_parquet_sink() {
     exit 1
   fi
   cd "${test_dir}"
-  mkdir omrs
-  print_message "Finding number of patients, encounters and obs in openmrs server"
-  curl -L -X GET -u admin:Admin123 --connect-timeout 5 --max-time 20 \
-    http://localhost:8099/openmrs/ws/fhir2/R4/Patient/ 2>/dev/null >>./omrs/patients.json
-  total_test_patients=$(jq '.total' ./omrs/patients.json)
-  print_message "Total openmrs test patients ---> ${total_test_patients}"
-  curl -L -X GET -u admin:Admin123 --connect-timeout 5 --max-time 20 \
-    http://localhost:8099/openmrs/ws/fhir2/R4/Encounter/ 2>/dev/null >>./omrs/encounters.json
-  total_test_encounters=$(jq '.total' ./omrs/encounters.json)
-  print_message "Total openmrs test encounters ---> ${total_test_encounters}"
-  curl -L -X GET -u admin:Admin123 --connect-timeout 5 --max-time 20 \
-    http://localhost:8099/openmrs/ws/fhir2/R4/Observation/ 2>/dev/null >>./omrs/obs.json
-  total_test_obs=$(jq '.total' ./omrs/obs.json)
-  print_message "Total openmrs test obs ---> ${total_test_obs}"
-  print_message "Counting number of patients, encounters and obs sinked to parquet files"
-  total_patients_streamed=$(java -jar ./parquet-tools-1.11.1.jar rowcount ./Patient/ | awk '{print $3}')
-  print_message "Total patients synced to parquet ---> ${total_patients_streamed}"
-  total_encounters_streamed=$(java -jar ./parquet-tools-1.11.1.jar rowcount ./Encounter/ | awk '{print $3}')
-  print_message "Total encounters synced to parquet ---> ${total_encounters_streamed}"
-  total_obs_streamed=$(java -jar ./parquet-tools-1.11.1.jar rowcount ./Observation/ | awk '{print $3}')
-  print_message "Total obs synced to parquet ---> ${total_obs_streamed}"
+
+  validate_counts
```

Comment:
I think the reason that the tests fail after this refactoring is because you have moved calculation of these variables (e.g., `total_test_patients`) to `validate_counts` but you are doing the actual verification outside that function (i.e., next line).

Reply:
Instead of the fix that you have done, couldn't we move the actual verification inside the `validate_counts` function?

Reply:
i realized that the fhir sink verification is different for both batch and stream modes hence could bring the code in the `common.sh` which keeps common peaces of code in both scripts forexample in batch mode a summary count is done on the fhir server with `http://localhost:8098/fhir/Patient/?_summary=count` which must bring back all the patients count as a right response whereas in stream mode we are only query to confirm whether the exact single patient is in the fhir server

Reply:
You are already passing options to `validate_counts` to restrict your `Patient` search (line 58 of `common.sh`), right? So you already have the option to differentiate between streaming and batch or more generally what you want to be counted. If the issue is the `_summary` parameter, I believe you can use it in both streaming and bash, can't you?

The problem with your current approach is that the `validate_counts` have global variable side effects which are better to avoid as much as possible. For example, look at how `total_test_patients` is written/read throughout the code. It is written in `validate_counts` but it is also written in `batch-mode-tests.sh` (i.e., below at line 87). These will quickly become hard to track. So when I see this variable in the code, I have to figure what was the last function that wrote this.

I see no reason that you cannot hide these variables from leaking to the global scope, i.e., define all of them as `local`. Only use global variables as the last resort and if you do so, keep the "write" part of that variable in a single function (i.e., only read outside that function). But again, for this particular case I don't see any reason that we cannot define these as `local` with more refactoring.